homeassistant/components/iotawatt/sensor.py

```diff
@@ -206,13 +228,39 @@ def extra_state_attributes(self):
         attrs = {"type": data.getType()}
         if attrs["type"] == "Input":
             attrs["channel"] = data.getChannel()
+        if self._accumulating:
+            attrs[
+                ATTR_LAST_UPDATE
+            ] = self.coordinator.api.getLastUpdateTime().isoformat()
 
         return attrs
 
+    async def async_added_to_hass(self):
+        """Load the last known state value of the entity if the accumulated type."""
+        await super().async_added_to_hass()
+        if self._accumulating:
+            state = await self.async_get_last_state()
+            self._accumulated_value = 0.0
+            if state:
+                try:
+                    self._accumulated_value = float(state.state)
+                    if ATTR_LAST_UPDATE in state.attributes:
+                        self.coordinator.update_last_run(
+                            dt.parse_datetime(state.attributes.get(ATTR_LAST_UPDATE))
+                        )
+                except (ValueError) as err:
```

Comment:
Can we avoid the `try/except` block if we know the unexpected value?

Reply:
I'm not quite sure I follow. The value is unexpected so how can we know it before hand?

Reply:
It looks like you are trying to guard against invalid data in `state.attributes.get(ATTR_LAST_UPDATE)` being passed to `dt.parse_datetime` or `float(state.state)` not being parsable as float.  How is that expected to happen?

Reply:
I'm doing the same way the `integration` integration is doing.

I can't think of how it could happen, but seeing that integration handles that case, I followed the same logic. https://github.com/home-assistant/core/blob/b3181a0ab2d269b6d1cb4e63aa78746c8d89e1e5/homeassistant/components/integration/sensor.py#L132